From dd9104a09a8f40d463730365c491a192484555b5 Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dimitar@dinux.eu>
Date: Thu, 13 Oct 2016 21:32:08 +0300
Subject: [PATCH 08/25] ld: testsuite: Mark PRU as elf target that does not
 support shared libraries

2016-12-26  Dimitar Dimitrov <dimitar@dinux.eu>

binutils/

	* testsuite/lib/binutils-common.exp (is_elf_format): Return
	false for PRU.

ld/
	* testsuite/ld-elf/eh-frame-hdr.d: Disable for PRU.
	* testsuite/ld-elf/sec-to-seg.exp: Disable for PRU.
	* testsuite/lib/ld-lib.exp (check_shared_lib_support): No shared
	libraries are supported for PRU.

Signed-off-by: Dimitar Dimitrov <dimitar@dinux.eu>
---
 ld/testsuite/ld-elf/eh-frame-hdr.d | 2 +-
 ld/testsuite/ld-elf/sec-to-seg.exp | 1 +
 ld/testsuite/lib/ld-lib.exp        | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/ld/testsuite/ld-elf/eh-frame-hdr.d b/ld/testsuite/ld-elf/eh-frame-hdr.d
index f73d0e51e6..a76ac73866 100644
--- a/ld/testsuite/ld-elf/eh-frame-hdr.d
+++ b/ld/testsuite/ld-elf/eh-frame-hdr.d
@@ -2,7 +2,7 @@
 #ld: -e _start --eh-frame-hdr
 #objdump: -hw
 #target: cfi
-#xfail: avr*-*-* or1k*-*-elf or1k*-*-rtems* visium-*-*
+#xfail: avr*-*-* or1k*-*-elf or1k*-*-rtems* pru-*-* visium-*-*
 # These targets support CFI generation but not shared libraries.
 #...
   [0-9] .eh_frame_hdr 0*[12][048c] .*
diff --git a/ld/testsuite/ld-elf/sec-to-seg.exp b/ld/testsuite/ld-elf/sec-to-seg.exp
index 8847318aa7..af3d604e18 100644
--- a/ld/testsuite/ld-elf/sec-to-seg.exp
+++ b/ld/testsuite/ld-elf/sec-to-seg.exp
@@ -87,6 +87,7 @@ if {    [istarget avr-*-*]
      || [istarget moxie-*-*]
      || [istarget msp430-*-*]
      || [istarget mt-*-*]
+     || [istarget pru-*-*]
      || [istarget visium-*-*]
     } {
     set B_test_same_seg 0
diff --git a/ld/testsuite/lib/ld-lib.exp b/ld/testsuite/lib/ld-lib.exp
index a2be49b25f..74dc998f14 100644
--- a/ld/testsuite/lib/ld-lib.exp
+++ b/ld/testsuite/lib/ld-lib.exp
@@ -1777,6 +1777,7 @@ proc check_shared_lib_support { } {
 	 && ![istarget nds32*-*-*]
 	 && ![istarget or1k*-*-*]
 	 && ![istarget pj-*-*]
+	 && ![istarget pru-*-*]
 	 && ![istarget rl78-*-*]
 	 && ![istarget rx-*-*]
 	 && ![istarget spu-*-*]
-- 
2.11.0

